### PR TITLE
Auth redirect

### PR DESCRIPTION
**Pull request by kristian240**
Implement a section where `AuthRedirect` component is described.

### CODE REVIEW

**Review by isBatak — APPROVED** [review]

File: React/Recipes/Session Handling.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+   */
+  onLogout?(): void;
+  /**
+   * on login success callback
+   */
+  onLogin?(session: SessionModel): void;
+}
+
+export const useAuth = ({ onLogin, onLogout, ...config }: IUseAuthOptions = {}) => {
+  const datx = useDatx();
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:27]: You are missing `useDatx` import

File: React/Recipes/Session Handling.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    }
+
+    return session;
+  }, [state, datx, onLogout]);
+
+  return { state, login, logout };
+};
+```
+
+This hook will expose session data and basic session handlers: `login` and `logout. Now, this hook can use this in any component to get session data.
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:62]:
```suggestion
This hook will expose session data and basic session handlers: `login` and `logout`. Now, this hook can be used in any component to get session data.
```

File: React/Recipes/Session Handling.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  return null;
+};
+```
+
+If we now implement this in our example, this now looks like this.
+
+```tsx
+const SomePrivatePage = () => {
+  return (
+    <>
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:175]: You don't need Fragment here, and we agreed on `<Fragment>` syntax

File: React/Recipes/Session Handling.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  ifFound?: boolean;
+  /**
+   * Callback that will trigger a redirect if true is returned.
+   * Useful when you need to redirect base on some attribute, e.g. if user is not admin
+   */
+  condition?(session: SessionModel): boolean;
+}
+
+const AuthRedirect: FC<IAuthRedirectProps> = ({ to, ifFound, condition }) => {
+  const { state } = useAuth();
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:133]:
```suggestion
  const { state: { data, isValidating, error } } = useAuth();
```

File: React/Recipes/Session Handling.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+      return;
+    }
+
+    const shouldRedirect = (ifFound && state.data) || (!ifFound && !state.data);
+
+    if (shouldRedirect) {
+      router.push(to);
+    }
+  }, [state.data, state.isValidating, state.error, to, condition, ifFound, router]);
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:163]:
```suggestion
  }, [data, isValidating, error, to, condition, ifFound, router]);
```

File: React/Recipes/Session Handling.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        router.push(to);
+      }
+
+      return;
+    }
+
+    const shouldRedirect = (ifFound && state.data) || (!ifFound && !state.data);
+
+    if (shouldRedirect) {
+      router.push(to);
```
Comment (isBatak) [inline React/Recipes/Session Handling.md:161]: You are missing `useRouter` at the beginning of component

**Review by DarkoKukovec — APPROVED** [review]

**Pull request merged**